### PR TITLE
Load full uniform in app-shell

### PR DESCRIPTION
**Pull request by alshakero**
5 seconds review. I tested with UniformDocs and all looks good with this app-shell (viewed by view-source):

```html
<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>UniformDocs</title>

    <script src="/sys/webcomponentsjs/webcomponents-lite.js"></script>
    <link rel="import" href="/sys/polymer/polymer.html">
    <link rel="import" href="/sys/starcounter.html">
    <link rel="stylesheet" href="/sys/uniform.css/dist/uniform.css">    
    <link rel="manifest">
</head>
<body>
    <dom-bind id="palindrom-root">
        <template>
            <starcounter-include view-model="{{model}}"></starcounter-include>
        </template>
    </dom-bind>
    
    <palindrom-client ref="palindrom-root" remote-url="/__default/AFEF5BB6513EFE1E7D000020"></palindrom-client>
</body>
</html>
```

Fixes: https://github.com/Starcounter/StarcounterClientFiles/issues/102

### CODE REVIEW

**Comment by warpech** [comment]
Please check the following:

1. [Uniform.css README](https://github.com/Starcounter/uniform.css/blob/24c3b1770e9351f4bcff73504cf063ad9cffa763/README.md) says that the recommended path to load the stylesheet is `/sys/uniform.css/uniform.css`. This is what all apps load. Please use that.
2. Many apps use `link rel=preload` to mitigate FOUC in shadow DOM, for example: https://github.com/Starcounter/UniformDocs/blob/63a6bfc7aff19eab72d17506b3e8c396c2a146c5/src/UniformDocs/wwwroot/UniformDocs/Components/DatepickerPage.html#L2. This is to solve point 1 of https://github.com/Starcounter/RebelsLounge/issues/244#issuecomment-369294149. Please check if putting `link rel=preload` directly in app shell's `<head>` solves this problem once and for all? Can `link rel=preload` be then removed from apps such as UniformDocs?

Edit: Originally in this comment there was a link to https://github.com/Starcounter/starcounter-include/issues/93 instead of point 1 of https://github.com/Starcounter/RebelsLounge/issues/244#issuecomment-369294149

**Comment by alshakero** [comment]
> This is what all apps load. Please use that.

Fixed

> Please check if putting link rel=preload directly in app shell's <head> solves this problem once and for all?

Sadly it does not. Multiple requests are still issued even with `preload` there in app-shell. 

image

**Comment by warpech** [comment]
Sorry, in my previous comment there was a wrong link which affected your interpretation of the problem.

I think that you should add `<link rel="preload" as="style" href="/sys/uniform.css/uniform.css">` above `<link rel="stylesheet" href="/sys/uniform.css/dist/underwear.css">`, because that allows to remove `preload` links from apps.

I know there are still multiple requests, but that is a known problem that is described in https://github.com/Starcounter/RebelsLounge/issues/244#issuecomment-369294149.

**Comment by alshakero** [comment]
> I think that you should add <link rel="preload" as="style" href="/sys/uniform.css/uniform.css"> above <link rel="stylesheet" href="/sys/uniform.css/dist/underwear.css">, because that allows to remove preload links from apps.

That is what I did, and I removed `preload` from UI Kit page in UniformDocs to test, enabled network throttling and the FOUC was still there. I initially thought the redundant network requests would explain the FOUC, but I guess my point stil stands even after the correct understanding of the problem. 

If adding a preload in the app-shell doesn't mitigate the FOUC nor the number of requests, what's the point of it?

**Comment by warpech** [comment]
The point of it is that `preload` makes a smaller FOUC than the lack of `preload`. Please read https://github.com/Starcounter/RebelsLounge/issues/244#issuecomment-369294149 and the preceding comments.

**Comment by warpech** [comment]
I just spend an hour or so, trying to see if there is any deduping effect of `preload` in `<head>`, but I can't find such.

Stylesheets seem to be deduped anyway in Chrome even without `preload`: https://next.plnkr.co/edit/99fFPSkg7DPoQL4F

@tomalec do you know how can this be explained?

It seems @alshakero is right that `preload` doesn't give any benefit of additional deduping.

**Comment by tomalec** [comment]
As far as I remember we were adding `link rel="preload"` in head (light dom) for styles loaded in shadow, not to dedupe, but to have a chance of loading the style before shadow root is attached => reducing the overall FOUC, by removing one of many reasons in some cases. As we do now load the style synchronously, directly in head, it gives even better results. As uniform will be there for sure for all `sc-include`'s composition. (but it's still one of many reasons of FOUC)

Such construct 
```html
 <link rel="preload" href="/sys/uniform.css/uniform.css">
 <link rel="stylesheet" href="/sys/uniform.css/uniform.css">
```

makes no benefit here. I believe it should be just `<link rel="stylesheet" href="/sys/uniform.css/uniform.css">`. It is cheaper as it is one element less, it and preload just before regular blocking script gives no benefit.

**Comment by tomalec** [comment]
Sorry as it seems I missed that in the review of https://github.com/Starcounter/uniform.css/pull/58#pullrequestreview-100107809 as the README note seems to be incorrect.

BTW, I don't think it is anything regarding network deduping we should worry about that much, as browser knows how to deal with it. I think even though browser may cache the resource, it needs to manifest new request in the network tab, as the stylesheet is parsed over and over as it's inserted later in the tree (=> may overwrite other existing rules).

The problem of sharing the same stylesheet resource and multiple `<style>` elements for custom element's shadow root is being discussed on Web Platform https://github.com/w3c/webcomponents/issues/468

I believe reusing same stylesheets among different shadow hosts is something we should give fro browser engine to optimize, as there is not much we can do, given the Uniform may or may not be in a particular partial's presentation.
The problem of same stylesheet used in multiple separated trees, and sharing it is general Web Platform/Browser problem, see

**Comment by tomalec** [comment]
Also to answer https://github.com/Starcounter/StarcounterClientFiles/pull/103#issuecomment-407662246

I rephrase it to 

> Please check if putting link `link rel="stylesheet" href="..uniform.css` directly in app shell's `<head>` solves the problem of FOUC of Uniform styles once and for all?

I don't think so. However, it solves on reason of FOUC once and for all, but only for Uniform styles, not the other external stylesheets loaded by compositions.

> Can link `rel=preload` be then removed from apps such as UniformDocs?

If those apps can assume the Uniform is loaded by the app shell, sure this preload can be removed. However, it still needs to be loaded in Shadow DOM composition (if it's being used there).

**Review by warpech — APPROVED** [review]
Based on the @tomalec's comment, I've reverted the preload. I think this is good to go

**Comment by alshakero** [comment]
I'll merge without releasing because I'm making another PR soon.
